tests/test_chimera.cpp

```diff
@@ -0,0 +1,408 @@
+/*
+    tests/test_chimera.cpp -- This demonstrates a hybrid usage of pybind11, where the
+    type caster returns a hand-rolled python object type rather than relying on the
+    natural python bindings.
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+#include <pybind11/pybind11.h>
+
+#include <deque>
+#include <map>
+#include <type_traits>
+#include <unordered_map>
+#include <cstring>
+#include <memory>
+
+#include "pybind11_tests.h"
+
+/// C++ type
+class Chimera {
```

Comment:
`class FreezableInt`
?
"Chimara" fits for the situation as a whole, but not for this type by itself.
Meaningful naming has a huge impact on readability, especially if this is meant to be a demo.

Reply:
Done.